src/App.js

```diff
@@ -17,9 +17,42 @@ import AddTodoForm from "./AddTodoForm";
 function App() {
   const [todoList, setTodoList] = React.useState([]);
   const [isLoading, setIsLoading] = React.useState(true);
+  const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
+
+  const fetchData = async () => {
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+      },
+    };
+    try {
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        const message = `Error: ${response.status}`;
+        throw new Error(message);
+      }
+      const data = await response.json();
+      const todos = data.records.map((todo) => {
+        const newTodo = {
+          id: todo.id,
+          title: todo.fields.title,
+        };
+
+        return newTodo;
+      });
+      setTodoList(todos);
+
+      setIsLoading(false);
+    } catch (error) {
+      console.log(error.message);
+    }
+  };
 
   React.useEffect(() => {
-    new Promise((resolve, reject) =>
+    fetchData();
```

Comment:
Nitpick:
This useEffect is missing `fetchData` as a dependency, however when defining functions within a component (App) body, the functions as dependencies must be wrapped in a [useCallback](https://react.dev/reference/react/useCallback) to avoid re-renders. Therefore, I'd recommend that you...

either define the `fetchData` within the useEffect body and invoke it:
```js
React.useEffect(() => {
  const fetchData = async () => { ... }
  fetchData();
}, []);
```
OR, wrap `fetchData` with a `useCallback` function and add it a dependency:
```js
const fetchData = React.useCallback(async () => { ... }, []);

useEffect(() => {
  fetchData();
}, [fetchData]);
```